pkg/rancher-desktop/store/preferences.ts

```diff
@@ -156,8 +156,10 @@ export const actions = {
     );
     commit('SET_PREFERENCES', newPreferences);
   },
-  setWslIntegrations({ commit }: PrefActionContext, integrations: { [distribution: string]: string | boolean}) {
-    commit('SET_WSL_INTEGRATIONS', integrations);
+  setWslIntegrations({ commit, state }: PrefActionContext, integrations: { [distribution: string]: string | boolean}) {
+    const updatedIntegrations = _.merge(_.cloneDeep(integrations), state.wslIntegrations);
```

Comment:
This can also be expressed as `_.merge({}, integrations, state.…` (because it only mutates the first element), I believe.